File: app/src/androidTest/java/com/xmartlabs/template/TestApplication.java
package com.xmartlabs.template;

import com.xmartlabs.template.module.AndroidModule;
import com.xmartlabs.template.module.MockRestServiceModule;

import bullet.ObjectGraph;

public class TestApplication extends BaseProjectApplication {
  @Override
  protected ApplicationComponent createComponent() {
    return DaggerTestComponent.builder()
        .coreAndroidModule(new AndroidModule(this))
        .restServiceModule(new MockRestServiceModule())
        .build();
  }

  @Override
  protected ObjectGraph createBullet(ApplicationComponent component) {
    return new BulletTestComponent((TestComponent) component);
  }
}

File: app/src/androidTest/java/com/xmartlabs/template/TestApplication.kt
package com.xmartlabs.template

import com.xmartlabs.template.module.AndroidModule
import com.xmartlabs.template.module.MockRestServiceModule

class TestApplication : BaseProjectApplication() {
  override fun createComponent() = DaggerTestComponent.builder()
      .coreAndroidModule(AndroidModule(this))
      .restServiceModule(MockRestServiceModule())
      .build()

  override fun createBullet(component: ApplicationComponent) = BulletTestComponent(component as TestComponent)
}
